### PR TITLE
feat: added AsideHeaderContext

### PR DESCRIPTION
**Pull request by sunduckcow**

### CODE REVIEW

File: src/components/Logo/Logo.tsx [hunk 1]
```diff
@@ -4,16 +4,14 @@ import {LogoProps} from '../types';
...
 
 import './Logo.scss';
+import {useAsideHeaderContext} from '../AsideHeader/AsideHeader';
 
 const b = block('logo');
 
-interface LogoInnerProps extends LogoProps {
-    compact: boolean;
-}
+interface LogoInnerProps extends LogoProps {}
```
Comment (Lunory) [inline src/components/Logo/Logo.tsx:11]: Remove useless type please

**Comment by Lunory** [comment]
FooterItem and ActionBar stories don't work
https://preview.gravity-ui.com/navigation/79/?path=/story/components-actionbar--single-section

File: src/components/AsideHeader/AsideHeader.tsx [hunk 2]
```diff
@@ -52,6 +52,20 @@ export interface AsideHeaderProps
 
 type AsideHeaderInnerProps = AsideHeaderGeneralProps & AsideHeaderDefaultProps;
 
+interface AsideHeaderContextType {
+    compact: boolean;
+    size: number;
+}
+
+const AsideHeaderContext = React.createContext<AsideHeaderContextType>({
```
Comment (DarkGenius) [inline src/components/AsideHeader/AsideHeader.tsx:60]: Let's move the context to a separate file

File: src/components/AsideHeader/AsideHeader.tsx [hunk 3]
```diff
@@ -52,6 +52,20 @@ export interface AsideHeaderProps
...
+    compact: false,
+    size: ASIDE_HEADER_COMPACT_WIDTH,
+});
+
+export const AsideHeaderContextProvider = AsideHeaderContext.Provider;
```
Comment (DarkGenius) [inline src/components/AsideHeader/AsideHeader.tsx:65]: Check these exports in `src/components/index.ts`

File: src/components/AsideHeader/AsideHeader.tsx [hunk 4]
```diff
@@ -17,6 +17,7 @@ import controlMenuButtonIcon from '../../../assets/icons/control-menu-button.svg
 import headerDividerCollapsedIcon from '../../../assets/icons/divider-collapsed.svg';
 
 import './AsideHeader.scss';
+import {AsideHeaderContextProvider} from './AsideHeaderContext';
```
Comment (DarkGenius) [inline src/components/AsideHeader/AsideHeader.tsx:20]: Let's move the import up

File: src/components/Logo/Logo.tsx [hunk 5]
```diff
@@ -4,16 +4,12 @@ import {LogoProps} from '../types';
 import {Button, Icon} from '@gravity-ui/uikit';
 
 import './Logo.scss';
+import {useAsideHeaderContext} from '../AsideHeader/AsideHeaderContext';
```
Comment (DarkGenius) [inline src/components/Logo/Logo.tsx:7]: Up

File: src/components/FooterItem/__stories__/FooterItem.stories.tsx [hunk 6]
```diff
@@ -7,6 +7,7 @@ import {ASIDE_HEADER_COMPACT_WIDTH, ASIDE_HEADER_EXPANDED_WIDTH} from '../../con
 import settingsIcon from '../../../../.storybook/assets/settings.svg';
 
 import './FooterItemShowcase.scss';
+import {AsideHeaderContextProvider} from '../../AsideHeader/AsideHeaderContext';
```
Comment (DarkGenius) [inline src/components/FooterItem/__stories__/FooterItem.stories.tsx:10]: Up

File: src/components/CompositeBar/Item/Item.tsx [hunk 7]
```diff
@@ -14,6 +14,7 @@ import {
 } from '../constants';
 
 import './Item.scss';
+import {useAsideHeaderContext} from '../../AsideHeader/AsideHeaderContext';
```
Comment (DarkGenius) [inline src/components/CompositeBar/Item/Item.tsx:16]: Up

**Review by Lunory — APPROVED** [review]

**Review by DarkGenius — APPROVED** [review]

**Pull request merged**